server/src/app/api/schemas.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2021, Pyronear contributors.
+
+# This program is licensed under the GNU Affero General Public License version 3.
+# See LICENSE or go to <https://www.gnu.org/licenses/agpl-3.0.txt> for full license details.
+
+
+from datetime import datetime
+from pydantic import BaseModel, Field, validator
+
+
+# Template classes
+class _CreatedAt(BaseModel):
+    created_at: datetime = None
+
+    @staticmethod
+    @validator('created_at', pre=True, always=True)
+    def default_ts_created(v):
+        return v or datetime.utcnow()
+
+
+class _Id(BaseModel):
+    id: int = Field(..., gt=0)
+
+
+# Metrics
+class MetricIn(_CreatedAt, _Id):
+    cpu_temperature_C: float = Field(None, example=17.3)
+    mem_available_GB: float = Field(None, example=1.23)
+    cpu_usage_percent: float = Field(None, example=51.8)
+
+
+class MetricOut(MetricIn):
+    pass
+
+
+class InferenceOut(_CreatedAt, _Id):
+    pass
```

Comment:
I'm not following? If we don't implement the inference route, let's remove it completely but I doubt the output schema is correct :sweat_smile:

Reply:
See discussion with inference route, as long as inference takes place in background, the output does not matter so much (but you are right, the schema is not correct, and might be better to at least return success/fail of the post (but this info is already in the status_code of the reponse). 

Or maybe, no output data ? thus, not InferenceOut class

Reply:
Alright but then let's not return anything :)

Here you define a schema with two fields : `created_at` and `id`. The `id` field is set to a fixed value (so it's not needed) and the created_at isn't used so the route will return a 422 status code :sweat_smile:

Reply:
If the inference run as a background task, what should we return to the device client ?

Reply:
An empty payload since we don't even know if the background task will succeed. We have very limited information to communicate (simply that we attempted launching a background task)